core/src/main/scala/org/broadinstitute/workbench/hamm/JsonCodec.scala

```diff
@@ -1,34 +1,77 @@
 package org.broadinstitute.workbench.hamm
 
+
+import java.time.format.DateTimeFormatter
+import java.time.temporal.{ChronoField, Temporal, TemporalAccessor}
+import java.text.SimpleDateFormat
+import java.time.Instant
+
 import cats.implicits._
-import io.circe.Decoder
+import io.circe.{Decoder, DecodingFailure}
+
+import scala.concurrent.duration.{Duration, FiniteDuration}
 
 object JsonCodec {
+
+  val formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
   implicit val cpuNumberDecoder: Decoder[CpuNumber] = Decoder.decodeString.emap(s => Either.catchNonFatal(s.toInt).leftMap(_.getMessage).map(CpuNumber))
   implicit val bootDiskSizeGbDecoder: Decoder[BootDiskSizeGb] = Decoder.decodeString.emap(x => Either.catchNonFatal(x.toInt).leftMap(_.getMessage).map(BootDiskSizeGb))
-  implicit val preemptibleDecoder: Decoder[Preemptible] = Decoder.decodeString.emap{
-    s =>
-      Either.catchNonFatal(s.toInt).leftMap(_.getMessage).map(Preemptible)
-  }
+
   implicit val diskNameDecoder: Decoder[Disks] = Decoder.decodeString.emap{
     str =>
       // sample value for str: `local-disk 1 HDD`
       for{
         array <- Either.catchNonFatal(str.split(" ")).leftMap(_.getMessage)
         size <- Either.catchNonFatal(array(1).toInt).leftMap(_.getMessage)
-      } yield Disks(DiskName(array(0)), DiskSize(size), DiskType(array(2)))
+      } yield Disks(DiskName(array(0)), DiskSize(size), DiskType.stringToDiskType(array(2)))
+  }
+
+
+  implicit val runtimeAttributesDecoder: Decoder[RuntimeAttributes] = Decoder.instance {
+    cursor =>
+      for {
+        cpuNumber <- cursor.downField("cpu").as[Int]
+        disks <- cursor.downField("disks").as[Disks]
```

Comment:
should this be `List[Disk]`?

Reply:
I chose not to handle calls with multiple disks in this PR but added a ToDo in the code (and I should create a ticket for it)